File: javascript/prime-factors/_iterations/1/prime-factors.js
export function primeFactors(n) {
    let remain = n;
    let factor = 2;
    const factors = [];
    while (remain > 1) {
        if (remain % factor == 0) {
            factors.push(factor);
            remain /= factor;
        } else {
            factor++;
        }
    }
    return factors;
}

File: r/prime-factors/_iterations/1/prime-factors.R
prime_factors <- function(number) {
  factor <- 2
  factors <- c()
  while (number > 1) {
    if (number %% factor == 0) {
      number <- number / factor;
      factors <- c(factors, factor)
    } else {
      factor <- factor + 1
    }
  }
  factors
}
